tests/openvino/test_training.py

```diff
@@ -116,317 +98,737 @@ class OVTrainerTestDescriptor:
+                padding="max_length",
+                max_length=max_length,
+                truncation=True,
+                return_tensors="pt",
+            )
+            result["labels"] = examples["label"]
+            return result
 
-        self.dataset = load_dataset("glue", "sst2")
-        self.train_dataset = self.dataset["train"].sort("sentence").select(range(8)).map(tokenizer_fn, batched=True)
-        self.eval_dataset = (
-            self.dataset["validation"].sort("sentence").select(range(4)).map(tokenizer_fn, batched=True)
-        )
-        self.metric = evaluate.load("glue", "sst2")
-        self.compute_metric = lambda p: self.metric.compute(
-            predictions=np.argmax(p.predictions, axis=1), references=p.label_ids
-        )
+        self.data_transform = data_transform
+        self.train_dataset = self.dataset["train"].select(range(8))
+        self.eval_dataset = self.dataset["validation"].select(range(4))
+        self.train_dataset.set_transform(data_transform)
+        self.eval_dataset.set_transform(data_transform)
+        self.data_collator = None
 
-    def get_nncf_config_with_overflow_fix_override(
-        self, nncf_compression_config: Union[List[Dict], Dict, None], value: str = "enable"
-    ):
-        overrided_config = deepcopy(nncf_compression_config)
-        quantization_config = None
-        if isinstance(overrided_config, list):
-            for config in overrided_config:
-                if config["algorithm"] == "quantization":
-                    quantization_config = config
-                    break
-        elif isinstance(overrided_config, dict):
-            if overrided_config["algorithm"] == "quantization":
-                quantization_config = overrided_config
-        if quantization_config is not None:
-            quantization_config["overflow_fix"] = value
-        return overrided_config
+    def check_ovmodel_output_equals_torch_output(self, ovmodel, torch_model):
+        torch_model = torch_model.eval()
+        for batch_size in [1, 4]:
+            for seq_length in [16, 89, 128]:
+                self.trainer.args.per_device_eval_batch_size = batch_size
+                dataset = deepcopy(self.eval_dataset)
+                dataset.set_transform(partial(self.data_transform, max_length=seq_length))
+                for inputs in self.trainer.get_eval_dataloader(dataset):
+                    self.assertSequenceEqual(inputs["input_ids"].shape, [batch_size, seq_length])
+                    ovmodel_outputs = ovmodel(**inputs)
+                    self.assertIn("logits", ovmodel_outputs)
+                    ovmodel_logits = ovmodel_outputs.logits
+                    with torch.no_grad():
+                        torch_logits = torch_model(**inputs).logits
+                    torch.testing.assert_close(
+                        torch.softmax(ovmodel_logits, dim=-1),
+                        torch.softmax(torch_logits, dim=-1),
```

Comment:
Could this be replaced with :
```suggestion
                        ovmodel_logits,
                        torch_logits,
```

Reply:
The softmax here is intended to amplify the logits so that atol=1e-4 can pass.

I check the model logits and find that sometimes the values are very close to zero, e.g., 0.0071. In **quantization** cases where higher error is expected than fp32 models, the original logit values are easy to accidentally fail at atol=1e-4 (e.g., 0.0071 vs. 0.0073).

In the latest commit I have deleted the softmax, but I also have to increase atol to 1e-3 to pass the tests (see latest checks here， ignoring the INC part which should not our problem). Is that acceptable? I see some other tests are also using 1e-3, e.g.,
https://github.com/huggingface/optimum-intel/blob/844be9adc9eb3bea8f87c22fa9112a4fb3a31d88/tests/openvino/test_modeling.py#L646

Reply:
A tolerance of 1e-3 sounds reasonable, thanks a lot for updating it !